### PR TITLE
Sync release changes with remake-4-3...

### PR DESCRIPTION
**Pull request by rocky**
Mostly removing/updating debugger help doc changes and updating the
copyleft.

Ignoring files and directories only in the remake-4-3 branch.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,15 +33,20 @@
 /configh.dos
 /configure
 /debian/remake.debhelper.log
+/dist
+/docs
+/lib
```
Comment (boretom) [inline .gitignore:38]: What is the idea of putting these directories into `.gitignore`? There are plenty of needed files in there, no?

**Review by boretom — APPROVED** [review]
Just my comment about `.gitignore`, fine otherwise

File: .gitignore (hunk 1 above)
Comment (rocky) [reply .gitignore:38]: In remake-4-2 these directories don't exist, so they are ignored. 

However if you start out in remake-4-3 then switch to remake-4-2, even though the files in the directories are gone, the directories themselves remain. 

I guess this is just the weird way git works.

**Comment by rocky** [comment]
So how do you feel about releasing remake-4-3 on March 15?  

Of course you have the means to make a dist tarball to test out. That's what I've been testing on.

Anything to report on your end? 

That is the one I'd most like to see get out there. remake-4-2-1 is optional. 

In either event though, we can always put out releases after this if something goes wrong. 

In a day or so after 4.3+1.5 goes out, I'll merge in the doxygen stuff. 

For 4.3 if you want to put the debugger stuff in a library by all means do it.  I think this isn't a too difficult task and if you have questions just ask. You can either read about it or look at how gnulib is currently done.

However the debugger library code should not be added to libgnu.a, (`-lgnu`) but should be own thing like  libdebugger.a (`-ldebugger`) or whatever you want to call it as long as it is different.

File: .gitignore (hunk 1 above)
Comment (boretom) [reply .gitignore:38]: I see, I tend to forget that 4.2 doesn't have the source files in `src`, you're right of course.

**Comment by boretom** [comment]
>So how do you feel about releasing remake-4-3 on March 15?

Sorry, didn't answer that, yes. I'm all for it of course. `make dist` works for me on macOS & FreeBSD.
It also works on OpenBSD but I'll need to write down what is required. `texinfo` got be compiled locally since the version with OpenBSD seems to be too old. But I don't think that is a show stopper.

> For 4.3 if you want to put the debugger stuff in a library by all means do it.

I haven't looked into it, have a few other things to organize. The whole Corona virus thing starts to get here too (I'm not in Switzerland right now) and I don't speak the local language well. Supposed to flight back in 2 weeks or so but then we're not supposed to go out except if you have to (and not sure I'm allowed to leave).
This means of course enough time for the library but I may not be as free in my mind as I should be.

And yes of course it should be in its own library, not `gnulib`. But what I didn't really understand is, does the interface change too, or "only" putting all the files in it's own library? Can this library be used outside of `make`/`remake`?

**Comment by rocky** [comment]
> does the interface change too, or "only" putting all the files in it's own library? 

I don't think the interface changes that much aside from functions many of which are in `remake/src/debugger/comand/*.h` which are currently marked "static". e.g. in `remake/src/debugger/command/print.h:

```
static debug_return_t
dbg_cmd_print(char *psz_args)
```

would become `extern`. Athough things could be jiggered so that the "init" functions like `dbg_cmd_print_init` could remain static such as by adding it to say `remake/src/debugger/command/help/print.h` (and then the directory name "help" would be a little misleading), probably the simplest is to make those be extern as well. 

> Can this library be used outside of make/remake?

The short definite-sounding answer is that to start out simple, assume no. 

This is a bit of an open-ended question and more about design. 

Once you put something in a library file then anyone who has access to the library can use it. However if the library isn't *installed* that makes it harder. And to use say `dbg_cmd_print()` you'd need the definition for `debug_return_t` so the header for that which is in `remake/src/debugger.h` would also need to be visible. And right now if you look in `remake/Makefile.am` you'll see that the only header definitions installed are those in `remake/src/gnuremake.h`. 

This still doesn't *stop* people from using things in the library if they really want to, just makes it harder.  In GNU Bash for example, I wanted to add some new optional built-in functions for a debugger. (They now are builtin so this is no longer needed). What you did in the `configure` scirpt step was indicate where the *source code* for bash was located and then it pulled in headers and even could pull in libraries or objects.  

if you do this, it doesn't really matter if this is in a _library_ or just in object files which presumably could be built sometime later too. 

Right now I don't see a *need* for something outside of remake to need access to the debugger code, but in the future there might be a need. But best to start out simple and assume nothing needs this and change that assumption and code when needed or desired. And since there is a workaround if this proves wrong, there is little harm except for invconvenience in the rare case, and the benefit is that it's easier to get going assuming this is internal initially.

**Pull request merged**